tools/interu/src/config/mod.rs

```diff
@@ -0,0 +1,186 @@
+            .get(profile_name)
+            .context(UnknownProfileNameSnafu { profile_name })?;
+
+        // Next, lookup the runner ref based on the profile strategy
+        let runner_ref = match &profile.strategy {
+            profile::Strategy::Weighted(options) => {
+                let weights: Vec<_> = options.weights.iter().map(|w| w.weight).collect();
+                let random_distribution = WeightedIndex::new(weights).unwrap();
+                let mut rng = thread_rng();
+
+                let index = random_distribution.sample(&mut rng);
+                let weight = options.weights.get(index).expect("always valid index");
+
+                &weight.runner
+            }
+            profile::Strategy::UseRunner(options) => &options.runner,
+        };
+
+        // Get the runner based on the runner ref
+        let runner = self.runners.get(runner_ref).unwrap();
+
+        // Get test options
+        let (test_parallelism, test_run, test_parameter) = profile.strategy.get_test_options();
+
+        // Convert our node groups to replicated node groups
+        let node_groups = runner
+            .node_groups
+            .clone()
+            .into_iter()
+            .map(|ng| ReplicatedNodeGroup::try_from(ng, &instances, &runner.platform.distribution))
+            .collect::<Result<Vec<_>, ConvertNodeGroupError>>()
+            .context(ConvertNodeGroupSnafu)?;
+
+        Ok(Parameters {
+            kubernetes_distribution: runner.platform.distribution.clone(),
+            kubernetes_version: runner.platform.version.clone(),
+            test_parameter: test_parameter.to_owned(),
+            test_run: test_run.clone(),
+            test_parallelism,
+            node_groups,
+        })
+    }
+}
+
+#[derive(Debug)]
+pub struct Parameters {
+    kubernetes_distribution: Distribution,
+    kubernetes_version: String,
+
+    node_groups: Vec<ReplicatedNodeGroup>,
+
+    test_parallelism: usize,
+    test_parameter: String,
+    test_run: TestRun,
+}
+
+impl Display for Parameters {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let Self {
+            kubernetes_distribution,
+            kubernetes_version,
+            node_groups,
+            test_parallelism,
+            test_parameter,
+            test_run,
+        } = self;
+
+        write!(f, "KUBERNETES_DISTRIBUTION={kubernetes_distribution}\n")?;
+        write!(f, "KUBERNETES_VERSION={kubernetes_version}\n")?;
+
+        // See: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#multiline-strings
+        write!(
+            f,
+            "NODE_GROUPS<<EOF\n{node_groups}\nEOF\n",
+            node_groups = serde_yaml::to_string(&node_groups).expect("must be serializable")
+        )?;
+
+        write!(f, "TEST_PARALLELISM={test_parallelism}\n")?;
+        write!(f, "TEST_PARAMETER={test_parameter}\n")?;
+        write!(f, "TEST_RUN={test_run}\n")
```

Comment:
Just an idea, group all of the one-liners together.

```suggestion
        write!(f, "KUBERNETES_DISTRIBUTION={kubernetes_distribution}\n")?;
        write!(f, "KUBERNETES_VERSION={kubernetes_version}\n")?;
        write!(f, "TEST_PARALLELISM={test_parallelism}\n")?;
        write!(f, "TEST_PARAMETER={test_parameter}\n")?;
        write!(f, "TEST_RUN={test_run}\n")?;
        
        // See: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#multiline-strings
        write!(
            f,
            "NODE_GROUPS<<EOF\n{node_groups}\nEOF\n",
            node_groups = serde_yaml::to_string(&node_groups).expect("must be serializable")
        )
```

Reply:
I also think we should namespace the env vars, eg: `INTERU_`.

This would require the github action to be aware of that change.

Reply:
> I also think we should namespace the env vars, eg: INTERU_.

Yes, I was planning on doing that.

> This would require the github action to be aware of that change.

Yes, this change will be part of the upcoming PR which switches the action to use this new tool.